popt.h

```diff
@@ -336,7 +336,7 @@ int poptAddItem(poptContext con, poptItem newItem, int flags);
 /** \ingroup popt
  * Perform sanity checks on a file path.
  * @param fn		file name
- * @return		0 on OK, 1 on NOTOK.
+ * @return		1 on OK, 0 on NOTOK.
```

Comment:
Why is this return code being inverted? Don't we typically want non-zero codes for `NOTOK`?

Reply:
It's being inverted because for a function named that way, you expect a true/false return, not a result code. You only need to look at the original code to be convinced just how strong that expectation is: they remembered the documented return code for the first test for stat(), but the rest of the tests go along the natural boolean route.

Reply:
Hmm, so yeah the *documentation* should be updated to read something in style of "Test whether a path appears sane for config file use".

Reply:
Updated the API documentation to better describe what it does.

Reply:
I'm still not sure it makes sense to return 1 for OK and 0 for NOTOK. Generally APIs like this return non-zero for bad states, and this is the opposite...

Reply:
I would agree if the function was called poptSanityCheck() or such. But because of the function name, *this* API answers the question "is this file sane?" where you want 1 for TRUE and 0 for FALSE. I can certainly update OK -> TRUE and NOTOK -> FALSE in the doco.

Reply:
I think that's better, yes.

Reply:
Well, it doesn't matter either way, you can leave it as-is, if the end result is that zero is the "bad" value.

Reply:
Let me put it this way: this is just a bug fix.

The existing return codes for a function named this way (it's a bad name for a public API to begin with) are backwards and buggy (the broken implementation really proves that point), and because this is a public API we can't rename it without a soname bump which we really don't want at this point, so we're stuck with it. This is the best we can do to salvage the stupid API.

Reply:
Alright, you win. That's fair.